test/int/cluster_test.go

```diff
@@ -68,72 +69,86 @@ var _ = Describe("AtlasCluster", func() {
 		removeControllersAndNamespace()
 	})
 
+	doCommonChecks := func() {
+		Expect(createdCluster.Status.ConnectionStrings).NotTo(BeNil())
+		Expect(createdCluster.Status.ConnectionStrings.Standard).NotTo(BeNil())
+		Expect(createdCluster.Status.ConnectionStrings.StandardSrv).NotTo(BeNil())
+		Expect(createdCluster.Status.MongoDBVersion).NotTo(BeNil())
+		Expect(createdCluster.Status.MongoURIUpdated).NotTo(BeNil())
+		Expect(createdCluster.Status.StateName).To(Equal("IDLE"))
+		Expect(createdCluster.Status.Conditions).To(ConsistOf(testutil.MatchConditions(
+			status.TrueCondition(status.ClusterReadyType),
+			status.TrueCondition(status.ReadyType),
+		)))
+		Expect(createdCluster.Status.ObservedGeneration).To(Equal(createdCluster.Generation))
+	}
+
+	performUpdate := func() {
+		Expect(k8sClient.Update(context.Background(), createdCluster)).To(Succeed())
+
+		Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterUpdatingFunc()),
+			1200, interval).Should(BeTrue())
+
+		doCommonChecks()
+	}
+
 	Describe("Create/Update the cluster", func() {
 		It("Should Succeed", func() {
 			expectedCluster := testAtlasCluster(namespace.Name, "test-cluster", createdProject.Name)
 
-			By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(expectedCluster)))
+			By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(expectedCluster)), func() {
```

Comment:
[q] what is the advantage of having the logic in callbacks - is it some visual separation or anything else?

Reply:
Yes, that way we have different "test cases" instead of different parts of one "test case"